test/DynamoCoreWpfTests/PackageManager/PackageManagerSearchElementViewModelTests.cs

```diff
@@ -1,50 +1,97 @@
-            packageManagerSearchViewModel.RefreshAndSearchAsync();
-
-            List<PackageManagerSearchElement> packageManagerSearchElements = ViewModel.PackageManagerClientViewModel.ListAll();
-
-            PackageManagerSearchElementViewModel packageManagerSearchElementViewModel = new PackageManagerSearchElementViewModel
-            (
-                packageManagerSearchElements.First(),
-                false,
-                packageManagerSearchViewModel.UserAlreadyHasPackageInstalled
-            );
-
-            packageManagerSearchElementViewModel.RequestDownload += packageManagerSearchViewModel.PackageOnExecuted;
-
-            // Necessary to dismiss any MessageBox which appears during the installation..
-            var dlgMock = new Mock<MessageBoxService.IMessageBox>();
-            dlgMock.Setup(m => m.Show(It.IsAny<string>(), It.IsAny<string>(), It.Is<MessageBoxButton>(x => x == MessageBoxButton.OKCancel || x == MessageBoxButton.OK), It.IsAny<MessageBoxImage>()))
-                .Returns(MessageBoxResult.OK);
-            MessageBoxService.OverrideMessageBoxDuringTests(dlgMock.Object);
-
-            // Act
-            Assert.AreEqual(false, packageManagerSearchElementViewModel.IsInstalled);
-            packageManagerSearchElementViewModel.DownloadLatestCommand.Execute(null);
-            
-            // Assert
-            Assert.AreEqual(true, packageManagerSearchElementViewModel.IsInstalled);
+            var name1 = "non-duplicate";
+            var name2 = "duplicate";
+            var version = "1.0.0";
+
+            var mockGreg = new Mock<IGregClient>();
+            var clientmock = new Mock<PackageManagerClient>(mockGreg.Object, MockMaker.Empty<IPackageUploadBuilder>(), string.Empty);
+            var pmCVM = new Mock<PackageManagerClientViewModel>(ViewModel, clientmock.Object);
+
+            var ext = Model.GetPackageManagerExtension();
+            var loader = ext.PackageLoader;
+            loader.Add(new Package("", name2, version, ""));
+
+            var packageManagerSearchViewModel = new PackageManagerSearchViewModel(pmCVM.Object);
+            packageManagerSearchViewModel.RegisterTransientHandlers();
+
+            var newSE1 = new PackageManagerSearchElementViewModel(new PackageManagerSearchElement(new PackageHeader()
+            {
+                name = name1
+            }), false);
+
+            var newSE2 = new PackageManagerSearchElementViewModel(new PackageManagerSearchElement(new PackageHeader()
+            {
+                name = name2
+            }), false);
+
+            packageManagerSearchViewModel.AddToSearchResults(newSE1);
+            packageManagerSearchViewModel.AddToSearchResults(newSE2);
+
+            // Default CanInstall should be true
+            Assert.AreEqual(true, newSE1.CanInstall);
+            Assert.AreEqual(true, newSE2.CanInstall);
```

Comment:
How is this true? It is already added to packageloader in line 30, right?

Reply:
There is no special meaning to it. It just checks that the default value set by the PackageManagerSearchElementViewModel constructor is true.